conbench/tests/api/test_benchmarks.py

```diff
@@ -774,6 +774,21 @@ def test_create_allow_just_sha(self, client):
         location = "http://localhost/api/benchmarks/%s/" % new_id
         self.assert_201_created(response, _expected_entity(benchmark_result), location)
 
+    @pytest.mark.parametrize("pr_number", [12345678, "12345678", None, "", "<absent>"])
+    def test_create_allow_pr_number_variations(self, pr_number, client):
+        self.authenticate(client)
+        data = copy.deepcopy(self.valid_payload)
+        data["run_id"] = _uuid()
+        data["github"]["pr_number"] = pr_number
+        if pr_number == "<absent>":
+            data["github"].pop("pr_number")
```

Comment:
Would you mind adding a comment about what the behavior is here that we're expecting? It looks like in the `"<absent>"` case, we remove `"<absent>"` before we send it to the client? Or am I missing something? 

We would expect that that would error, right? Maybe we should make a separate test that tests and confirms that? Or we can leave it out if it's effectively the same thing as sending nothing?

Reply:
Yeah, I put that test here because not passing `pr_number` is documented as acceptable behavior. There's both the case Austin mentioned, plus the case where the user specifies `branch` directly (which is discouraged, but possible, and if for some reason someone had builds on a branch without a PR, the only way to submit it).

Reply:
That behavior makes sense — the thing I was confused by is _how_ this code maps onto that (TBF, I didn't get to there reading the code it looked more like a mistake or testing some strange behavior where we expect a user to send `"<absent>"` and wanted to catch that). 

Something like what you say here in a comment (maybe along with "the `"<absent>"` sentinel here  is used so that we can test that case where we don't send pr_number at all" if that's accurate) would be fantastic to make that less confusing

Reply:
Added a [somewhat verbose] comment to that effect